docs/ref/migration-operations.txt

```diff
@@ -365,6 +365,84 @@ It accepts two list of operations, and when asked to apply state will use the
 state list, and when asked to apply changes to the database will use the database
 list. Do not use this operation unless you're very sure you know what you're doing.
 
+Creating migrations that add unique fields
+===========================================
+
+Applying a migration that adds unique fields to existing rows will raise an error
+because the unique field will be generated with the same value added to all rows,
+breaking the ``unique`` constraint.
+
+Therefore, the following steps should be taken (in this example, we'll
+create a uuid field, so modify with appropriate ``default`` arg for your
+needs):
+
+* Add the field with ``default=uuid.uuid4`` and ``unique=True`` args.
+
+* Run the ``makemigrations`` command.
+
+* Edit the created migration file.
+
+* The migration class should look similar to this::
+
+    class Migration(migrations.Migration):
+
+        dependencies = [
+            ('myapp', '0003_auto_20150129_1705'),
+        ]
+
+        operations = [
+            migrations.AddField(
+                model_name='mymodel',
+                name='uuid',
+                field=models.UUIDField(max_length=32, unique=True, default=uuid.uuid4),
+            ),
+        ]
+
+* You will need to make three changes:
+
+    - Change the ``AddField`` operation to create an intermediary null
+      field.
+
+    - Add a RunPython operation to generate a UUID for each existing row.
+
+    - Add a second ``AmendField`` operation to change the field into
+      ``unique=True`` and ``null=False`` one.
+
+* The resulting migration should look similar to this::
+
+    import uuid
+
+    def gen_uuid(apps, schema_editor):
+        from myapp.models import Mymodel
+        for row in Mymodel.objects.all():
+            row.uuid = uuid.uuid4()
+            row.save()
+
+    class Migration(migrations.Migration):
+
+        dependencies = [
+            ('myapp', '0003_auto_20150129_1705'),
+        ]
+
+        operations = [
+            migrations.AddField(
+                model_name='mymodel',
+                name='uuid',
+                field=models.UUIDField(max_length=32, null=True),
+            ),
+
+            migrations.RunPython(gen_uuid),
```

Comment:
You could add a comment referring to ``RunPython.noop`` for a ``reverse_code`` to make the migration file reversible.